examples/lib/dynamic_outputs/pipeline.ex

```diff
@@ -0,0 +1,194 @@
+
+    0..5
+    |> Enum.each(fn output_number ->
+      five_seconds = 5_000
+
+      Process.send_after(
+        self(),
+        {:register_output, output_id(output_number)},
+        output_number * five_seconds
+      )
+    end)
+
+    {[spec: spec, spec: input_specs], %{sample_path: sample_path, inputs: [], outputs: []}}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {register, _id, vc_ctx},
+        :video_compositor,
+        _ctx,
+        state
+      )
+      when register == :input_registered or register == :output_registered do
+    {inputs, outputs} = inputs_outputs_ids(vc_ctx)
+    {update_scene_action(inputs, outputs), %{state | inputs: inputs, outputs: outputs}}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:new_output_stream, output_id, _vc_ctx},
+        :video_compositor,
+        _membrane_ctx,
+        state
+      ) do
+    Process.send_after(self(), {:remove_output, output_id}, 10_000)
+    {[spec: output_spec(output_id)], state}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:vc_request_response, req, %Req.Response{status: response_code, body: response_body},
+         _vc_ctx},
+        :video_compositor,
+        _membrane_ctx,
+        state
+      ) do
+    if response_code != 200 do
+      raise """
+      Request failed.
+      Request: `#{inspect(req)}.
+      Response code: #{response_code}.
+      Response body: #{inspect(response_body)}.
+      """
+    end
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_child_notification(notification, child, _ctx, state) do
+    Membrane.Logger.info(
+      "Received notification: #{inspect(notification)} from child: #{inspect(child)}."
+    )
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_info({:register_output, output_id}, _ctx, state) do
+    output_opt = %OutputOptions{
+      width: @output_width,
+      height: @output_height,
+      id: output_id
+    }
+
+    {[notify_child: {:video_compositor, {:register_output, output_opt}}], state}
+  end
+
+  @impl true
+  def handle_info(
+        {:remove_output, output_id},
+        _ctx,
+        state = %{inputs: inputs, outputs: outputs}
+      ) do
+    outputs = outputs |> Enum.reject(fn id -> id == output_id end)
+
+    children = [
+      {:output_parser, output_id},
+      {:output_decoder, output_id},
+      {:sdl_player, output_id}
+    ]
+
+    # Change the scene before removing the output.
+    # VideoCompositor forbids removing output used in the current scene.
+    {update_scene_action(inputs, outputs) ++ [remove_children: children],
+     %{state | outputs: outputs}}
+  end
+
+  @spec update_scene_action(list(VideoCompositor.input_id()), list(VideoCompositor.output_id())) ::
+          [Membrane.Pipeline.Action.notify_child()]
+  defp update_scene_action(input_ids, output_ids) do
+    update_scene_request =
+      if not Enum.empty?(output_ids) and not Enum.empty?(input_ids) > 0 do
```

Comment:
If any of the specified nodes are unused for composition, then VideoCompositor returns the `400` code in the request-response. VideoCompositor defines node unused, one it doesn't provide any texture used in the output composition (basically runs DFS from output nodes and checks if all nodes have been reached). When there are no outputs, then this code would still define a single node that can't be used, therefore VC would return an error. It can be avoided with a separate function, but it just moves the if statement to a different place.

Reply:
If I'm not mistaken, my suggestion would not change the behaviour of the code. But it's a minor thing, you can leave it as is